Sources/PostgresNIO/New/PostgresCodable.swift

```diff
@@ -2,29 +2,59 @@ import NIOCore
 import class Foundation.JSONEncoder
 import class Foundation.JSONDecoder
 
+/// A type that can encode itself to a Postgres wire binary representation.
+/// Dynamic types are types that don't have a known Postgres type at compile time. For example, a custom
+/// type that is created at runtime.
+public protocol PostgresDynamicTypeThrowingEncodable {
```

Comment:
```suggestion
public protocol PostgresThrowingDynamicTypeEncodable {
```
My reasoning for the different naming: `PostgresDynamicTypeThrowingEncodable` reads to me as "an encodable throwing a Postgres dynamic type", rather than "a throwing dynamically-typed encodable".